src/Guide/Api/Utils.hs

```diff
@@ -89,3 +102,47 @@ instance (HasSwagger api, KnownSymbol name, KnownSymbol desc) =>
              symbolVal (Proxy @desc))
             Nothing
      in toSwagger (Proxy @api) & applyTags [tag]
+
+-- | Servant request details, can be captured by adding `RequestDetails :>` to an API branch.
+data RequestDetails = RequestDetails
+  { rdIp        :: Maybe IP  -- request ip address
```

Comment:
Field comments have to start with `-- ^`, otherwise they won't be rendered by Haddock.